pilot/pkg/networking/util/util.go

```diff
@@ -540,7 +540,7 @@ func BuildLbEndpointMetadata(network, tlsMode, workloadname, namespace string, l
 		addIstioEndpointLabel(metadata, "network", &pstruct.Value{Kind: &pstruct.Value_StringValue{StringValue: network}})
 	}
 
-	if tlsMode != "" {
+	if tlsMode != "" && tlsMode != model.DisabledTLSModeLabel {
```

Comment:
did we still need this change as well? or was it part of the cluster part the was accidentally included?

Reply:
I don't think we 100% need it but it fixes an inconsistency in the logic for building metadata.